include/irr/asset/ICPUMeshBuffer.h

```diff
@@ -92,6 +97,39 @@ class ICPUMeshBuffer : public IMeshBuffer<ICPUBuffer, ICPUDescriptorSet, ICPURen
     {
         return m_descriptorSet.get();
     }
+	inline void setVertexAttrBuffer(uint16_t attribIndex, E_FORMAT format, uint32_t stride, uint32_t offset)
+	{
+		auto areParamsInValidRange = [&]()
+		{
+			auto isParamValid = [](const uint32_t &value, const uint32_t &maxRange)
+			{
+				if (value <= maxRange)
+					return true;
+				else
+				{
+					std::string irrlichWayToHandleErrors("error handle: out of range (" + std::to_string(value) + ")!\n"); // TODO there should be an Irrlich way to raport errors
+					return false;
+				}
+			};
+
+			std::vector<std::pair<uint32_t, uint32_t>> valuesAndAssignedMaxRanges({ std::make_pair(attribIndex, SVertexInputParams::MAX_VERTEX_ATTRIB_COUNT), std::make_pair(stride, TODO), std::make_pair(offset, TODO) });  // TODO find those constexpr
```

Comment:
@devshgraphicsprogramming  I have been looking for those constexpr, but _IRR_STATIC_INLINE_CONSTEXPR doesn't appear to have them and simmilar
is it directly opengl max const or have you created those yet in another place?

second thing I have to know your reporting errors system

Reply:
you can make the things you've marked as TODO, the max values for them are 2048 (not inclusive, so 2047 inclusive)

You have absolutely no permission to stick a random `vector` in the middle of a class as a member. Use a static array if you must.